test/fuzzing/atom.cpp

```diff
@@ -0,0 +1,182 @@
+#include "pms-utils/atom/atom.hpp"
+
+#include "pms-utils/atom/atom_parser.hpp"
+#include "pms-utils/misc/try_parse.hpp"
+#include "validate_utf8.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <cwchar>
+#include <format>
+#include <nanobind/nanobind.h>
+#include <optional>
+#include <ranges>
+#include <span>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+
+// the python header lacks any IWYU-awareness, and the individual headers are broken
+// put them AFTER the nanobind header, so that the dependant headers get included by it
+#include <cpython/initconfig.h>
+#include <pyerrors.h>
+#include <pylifecycle.h>
+
+// IWYU pragma: begin_keep
+#include <nanobind/stl/optional.h>
+#include <nanobind/stl/string.h>
+// IWYU pragma: end_keep
+
+namespace nb = nanobind;
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+    [[maybe_unused]] static const bool init = []() {
```

Comment:
What's `init` used for here? I don't see it used anywhere else, and a quick skim in the nanobind docs doesn't show anything about some global `std::function<void>`